tapir/coop/services/ResignMemberService.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+
+from tapir.accounts.models import TapirUser
+from tapir.coop.models import ResignedMembership, ShareOwnership
+from tapir.coop.forms import MembershipCancelForm
+from tapir.shifts.models import (
+    ShiftAttendanceTemplate,
+    ShiftAttendance,
+    ShiftAccountEntry,
+)
+from tapir.shifts.services.frozen_status_service import FrozenStatusService
+from tapir.shifts.models import ShiftUserData, ShiftAttendanceMode
+
+from tapir.utils.shortcuts import get_timezone_aware_datetime
+
+
+class ResignMemberService:
+    @staticmethod
+    def update_shifts_and_shares(
```

Comment:
We probably want to add the `@transaction.atomic` decorator, so that if an error happens in the middle of processing all the shifts and shares, the changes get rolled back to how it was before the function was called.
Otherwise we could end up with half the shares transfered but not the others and it's tricky to fix that.